### PR TITLE
Improving the router initialization checking logic: polling --> evented

### PR DESCRIPTION
**Pull request by adityamukho**
### Scope & Purpose

This PR improves the initialization checking code that is executed for each Backbone route. The existing version is a bit roundabout, using a timer to poll for the _ready_ state of the router. This is not ideal in an asynchronous programming model like JS, where the event loop ought to be leveraged instead. So, the polling timer has been replaced with a `Promise` that invokes the route code after the initialization is guaranteed to have finished.

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [ ] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [x] No backports required
- [ ] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket number:
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [x] This change is a trivial rework / code cleanup without any test coverage.
- [x] The behavior in this PR was *manually tested*
- [x] This change is already covered by existing tests, such as https://jenkins.arangodb.biz:3456/view/PR/job/arangodb-matrix-pr/
- [ ] This PR adds tests that were used to verify all changes:
  - [ ] Added new C++ **Unit tests**
  - [ ] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run: https://jenkins.arangodb.biz:3456/view/PR/job/arangodb-matrix-pr/16340/

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

File: js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js [hunk 1]
```diff
@@ -340,13 +313,14 @@
         });
 
         arangoHelper.initSigma();
-      }.bind(this);
+      }).bind(this);
 
-      $(window).resize(function () {
+      // TODO: Should a UI function be part of the router?
```
Comment (hkernbach) [inline js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js:318]: No it shouldn't :-) 
Let's get rid of those problems as soon as we can get rid of the Backbone router.
But I bet this problem will be solved anyway, as soon as we can deprecate Backbone views - at least the way how they are currently implemented.

File: js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js [hunk 2]
```diff
@@ -340,13 +313,14 @@
...
+      $(window).on('resize', function () {
         self.handleResize();
       });
 
-      $(window).scroll(function () {
+      $(window).on('scroll', function () {
         // self.handleScroll()
       });
```
Comment (hkernbach) [inline js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js:325]:
```suggestion
```

Comment (hkernbach) [reply js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js:325]: Can be removed completely, as it does not execute any code.

**Review by hkernbach — APPROVED** [review]
LGTM. Just minor things.
One side note for upcoming PR's. 

In general I like do formatting our code. But in this case here, it's hard to review because almost everything got reformatted. 

We should somehow solve this. One possible solution would be to reformat the whole code at once and push it to devel in a dedicated PR. 

This would make future reviews a lot more easier.

File: js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js (hunk 1 above)
Comment (adityamukho) [reply js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js:318]: Agreed. This issue will go away when the Backbone router is deprecated. No action needed for now 👍

File: js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js (hunk 2 above)
Comment (adityamukho) [reply js/apps/system/_admin/aardvark/APP/frontend/js/routers/router.js:325]: Can also remove the handler? I don't think it is used for anything else.

**Comment by adityamukho** [comment]
> LGTM. Better init solution as before ;) Just minor things.
> One side note for upcoming PR's.
> 
> In general I like do formatting our code. But in this case here, it's hard to review because almost everything got reformatted.
> 
> We should somehow solve this. One possible solution would be to reformat the whole code at once and push it to devel in a dedicated PR.
> 
> This would make future reviews a lot more easier.

Locally, I set my IDE to ignore whitespace changes. That makes it a lot easier to review. Not sure if GH has such an option though.

**Review by hkernbach — APPROVED** [review]

**Pull request merged**